entproto/cmd/protoc-gen-entgrpc/converter.go

```diff
@@ -136,13 +163,32 @@ func (g *serviceGenerator) renderToProto(conv *converter, varName, ident string)
 	}
 
 	switch {
-	case conv.toProtoTryConstructor.GoName != "":
-		// Returns: varName, err := ProtcoConstructor(entMember) with error handler
-		return fmt.Sprintf(`%s, err := %s(%s)
+	case conv.toProtoValuer != "" && conv.toProtoConstructor.GoName != "":
+		// Returns: varName, err := entMember.Value().(type) with error handling
+		return fmt.Sprintf(`%sValue, err := %s.Value()
 		if err != nil {
 			return nil, err
 		}
-		`, varName, g.QualifiedGoIdent(conv.toProtoTryConstructor), ident)
+
+		%sTyped, ok := %sValue.(%s)
+		if !ok {
+			return nil, %s("casting value to %s")
+		}
+
+		%s := %s(%sTyped)
+		`, varName, ident, varName, varName, conv.toProtoValuer, "%(newError)", conv.toProtoValuer, varName, g.QualifiedGoIdent(conv.toProtoConstructor), varName)
```

Comment:
these are getting pretty complex. i opted here to duplicate the logic rather than increase the conditional logics complexity (since it is hard to reason about).

i think as a followup we should improve the templating system to allow us to use them here. that way we can have a single way of templating rather than combining fmt and the imperative templating helper.